File: backend/app/schemas/agent.py
# Pydantic schemas for AI agent API
# Classes:
#   AgentQueryRequest: { question: str, session_id: str }
#   AgentChartPayload: { type: str (bar/pie/line/scatter), title: str, data: list[dict], x_key: str, y_key: str, color_key: str | None }
#   AgentQueryResponse: { answer: str, chart_data: AgentChartPayload | None }

File: frontend/src/types/agent.ts
export type ChartType = 'bar' | 'pie' | 'line' | 'scatter'

export interface AgentChartPayload {
  type: ChartType
  title: string
  data: Record<string, unknown>[]
  xKey: string
  yKey: string
  colorKey?: string
}

export interface AgentQueryRequest {
  question: string
  sessionId: string
}

export interface AgentQueryResponse {
  answer: string
  chartData?: AgentChartPayload
  htmlArtifact?: string
}
